src/backend/joanie/core/api.py

```diff
@@ -190,3 +206,130 @@ def delete(self, request, **kwargs):
             obj.delete()
             return Response(status=status.HTTP_204_NO_CONTENT)
         return Response(status=status.HTTP_400_BAD_REQUEST)
+
+
+class CreditCardView(generics.ListAPIView):
+    """
+    API view allows to get all credit cards or update or create a new one for a user.
+
+    GET /api/credit_cards/
+        Return list of all user credit cards saved
+        [
+            {
+                'id': uid,
+                'name': str,
+                'last_numbers': str, four last digits
+                'expiration_date': date, only "month/year" e.g. "09/22"
+                'main': bool,
+            },
+        ]
+
+    POST /api/credit_cards/ allows to register a new credit card with expected data:
+        - name: str, user can name his/her credit card
+        - card_number: str, credit card number e.g. 1111222233334444
+        - cryptogram: str, credit card cryptogram e.g. 222
+        - expiration_date: str, only "month/year" e.g. "09/22" is expected
+        - main: bool,
+
+    PUT /api/credit_cards/<credit_card_uid>/ allows to modify credit card name and main flag
+    with expected data:
+        - name: str,
+        - main: bool,
+
+    DELETE /api/credit_cards/<credit_card_uid>/ allows to remove credit card
+    """
+
+    permission_classes = [permissions.IsAuthenticated]
+    serializer_class = serializers.CreditCardModelSerializer
+    instance_field = "credit_card_uid"
+
+    def get_instance(self, **kwargs):
+        """Get credit card instance"""
+        return get_object_or_404(models.CreditCard, uid=kwargs[self.instance_field])
+
+    def get_queryset(self):
+        """Custom queryset to get user credit cards"""
+        user = models.User.objects.get_or_create(username=self.request.user.username)[0]
+        return user.creditcards.all()
+
+    def post(self, request):
```

Comment:
I'm surprised that this is done here. It's usually done in the frontend so that the card credentials don't go through your server. I would say we need a certification to receive handle such card credentials as they could be leaked by our proxies or logging chains, etc.
Ususally the frontend requests a token from the backend with which it can then call the payment service directly with the credentials.